std/experimental/color/rgb.d

```diff
@@ -0,0 +1,852 @@
+module std.experimental.color.rgb;
+
+import std.experimental.color;
+import std.experimental.color.xyz;
+
+import std.traits: isInstanceOf, isNumeric, isIntegral, isFloatingPoint, isSigned, isSomeChar, Unqual;
+import std.typetuple: TypeTuple;
+import std.typecons: tuple;
+
+@safe: pure: nothrow: @nogc:
+
+
+enum isRGB(T) = isInstanceOf!(RGB, T);
+
+enum RGBColorSpace
+{
+    sRGB,
+    sRGB_Gamma2_2,
+/+ TODO: these all need testing
```

Comment:
RGB colour space isn't about the gamma ramp, it's about the position of the colour axiis relative to a whitepoint in the XYZ colour space. RGB colour spaces all have different gamuts. The linearity of the encoding is just a trade-off between ability to sum luminances, and efficiency of encoding space relative to human luminance perception.